tools/update_extern.py

```diff
@@ -0,0 +1,84 @@
+"""
+Updates all the libraries in ``sunpy/extern``
+"""
+
+import os
+import json
+import shutil
+import urllib.request
+from parfive import Downloader
+from pathlib import Path
+from zipfile import ZipFile
+
+SUNPY_EXTERN_DIR = Path(__file__).parent.parent / "sunpy" / "extern"
+
+# "package_name": ["user", "repository", "path_to_file"]
+PACKAGES = {
+    "appdirs": ["ActiveState", "appdirs", "appdirs.py"],
+    "distro": ["python-distro", "distro", "src/distro/distro.py"],
+    "inflect": ["jaraco", "inflect", "inflect/__init__.py"],
+    "parse": ["r1chardj0n3s", "parse", "parse.py"],
+}
+
+
+def download_package(user: str, repo: str):
+    """
+    Download the latest version of package using Github release tags.
+
+    Args:
+        user: Github user name
+        repo: Github repository name
+    """
+    print(f"Checking {user}/{repo}")
+    response = urllib.request.urlopen(f"https://api.github.com/repos/{user}/{repo}")
+    if response.status != 200:
+        raise ValueError(f"{user}/{repo} does not exist.")
+
+    url = f"https://api.github.com/repos/{user}/{repo}/releases/latest"
+    response = urllib.request.urlopen(url)
+    if response.status != 200:
+        url = f"https://api.github.com/repos/ActiveState/appdirs/tags"
+        response = urllib.request.urlopen(url)
+        response = json.load(response)
+        version = response[0]["name"]
```

Comment:
Why is this here?

Reply:
Oh, I missed that

Reply:
I think they all have releases so the tags code isn't run. But we should probably keep it anyway.

Reply:
But this is only for one package? It isn't checking the tags for each project that I can see.

Maybe we should fallback to tags if no release is found for each project?

Reply:
It is doing that, just `f"https://api.github.com/repos/ActiveState/appdirs/tags"` should be `f"https://api.github.com/repos/{user}/{repo}/tags"`. I tried `raise ValueError()` in that code branch and the error wasn't raised so it found a release for all.

Reply:
If we need to fall back to tags, would using the tags not be a better place to start?